sale_blanket_order/models/sale_orders.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2018 ACSONE SA/NV
+# Copyright 2019 Eficent and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import date, timedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    blanket_order_id = fields.Many2one(
+        "sale.blanket.order",
+        string="Origin blanket order",
+        related="order_line.blanket_order_line.order_id",
+    )
+
+    @api.model
+    def _check_exchausted_blanket_order_line(self):
+        return any(
+            line.blanket_order_line.remaining_qty < 0.0 for line in self.order_line
+        )
+
+    def button_confirm(self):
```

Comment:
@nguyenminhchien thanks for the PR! Isn't this method called `action_confirm`? I cannot find the `button_confirm` action for `sale.order`.

Reply:
you're right, the correct one is `action_confirm`.
I just updated it, please have a look.